src/shader_recompiler/frontend/translate/translate.cpp

```diff
@@ -313,12 +309,69 @@ IR::U64F64 Translator::GetSrc64(const InstOperand& operand, bool force_flt) {
 }
 
 template <>
-IR::U64 Translator::GetSrc64(const InstOperand& operand, bool force_flt) {
-    return GetSrc64<IR::U64F64>(operand, force_flt);
+Translator::SrcAuto Translator::GetSrc<Translator::SrcAuto>(const InstOperand& operand) {
```

Comment:
Lots of complication here I think. Not sure why this needed but if you want to avoid moving GetSrc in header, you can make it a template and add special definition for U32/F32 here. Also make the default type in the template as IR::F32

Reply:
This SrcAuto is to make the type infer based on the return type. Having this wrapper, it can call the correct method using the cast operator to the expected type, otherwise, all GetSrc would require the template parameter as it would construct a new IR::Value from whatever type it returns, losing all the compile-time check